kpops/components/base_components/base_defaults_component.py

```diff
@@ -193,31 +190,44 @@ def defaults_from_yaml(path: Path, key: str) -> dict:
 
 
 def get_defaults_file_paths(
-    config: KpopsConfig, environment: str | None
-) -> tuple[Path, Path]:
+    pipeline_path: Path, config: KpopsConfig, environment: str | None
+) -> list[Path]:
     """Return the paths to the main and the environment defaults-files.
 
     The files need not exist, this function will only check if the dir set in
     `config.defaults_path` exists and return paths to the defaults files
     calculated from it. It is up to the caller to handle any false paths.
 
+    :param pipeline_path: The path to the pipeline.yaml file
     :param config: KPOps configuration
     :param environment: Environment
     :returns: The defaults files paths
     """
-    defaults_dir = Path(config.defaults_path).resolve()
-    main_default_file_path = defaults_dir / Path(
-        config.defaults_filename_prefix
-    ).with_suffix(".yaml")
-
-    environment_default_file_path = (
-        defaults_dir
-        / Path(f"{config.defaults_filename_prefix}_{environment}").with_suffix(".yaml")
-        if environment is not None
-        else main_default_file_path
-    )
+    default_paths = []
+
+    if not pipeline_path.is_file():
+        message = f"No pipeline.yaml found in directory: {pipeline_path.parent}"
```

Comment:
I would prefer
```suggestion
        message = f"{pipeline_path} is not a valid pipeline file."
```
the reason is that KPOps also supports `pipeline.json` which we use in one external project that's based on KPOps